### DISCUSSION TITLE
Comments on models

### DISCUSSION THREAD

**Discussion by Chewbaka69** [lanedirt/OGameX#117] [General]
Hey!

I just have a little question, i saw in models you put all the properties and method in comments and not in code.
Is there a reason, i am just interesting to know :)

**Comment by lanedirt** [comment]
Hi, the properties and methods in Models are generated by the laravel-ide-helper module: https://github.com/barryvdh/laravel-ide-helper. These comments are used exclusively for improved IDE autocompletion. I myself use PHPStorm from JetBrains where this works pretty good.

Whenever the database structure is modified these comments can be re-generated with this artisan command: 

`php artisan ide-helper:models`.

Laravel Eloquent models use PHP magic getter/setter methods by default for all properties/attributes. That's why they aren't defined in code like you normally would expect. Hence the reason why the ide-helper package exists in order to improve the developer flow.

**Reply by Chewbaka69** [reply]
Ooooooh because i use PHPStorm too, and i do not know this module, so good to know thanks a lot :)
